lib/telepoison.ex

```diff
@@ -206,29 +217,64 @@ defmodule Telepoison do
     Tracer.set_current_span(ctx)
   end
 
+  defp get_standard_attributes(request, host) do
+    [
+      {"http.method",
+       request.method
+       |> Atom.to_string()
+       |> String.upcase()},
+      {"http.url", request.url},
+      {"net.peer.name", host}
+    ]
+  end
+
+  defp get_attributes(opts) do
+    case get_defaults() do
+      {_, defaults} when is_list(defaults) ->
+        attributes = Keyword.get(opts, :ot_attributes, [])
+        Keyword.merge(attributes, defaults)
+
+      _ ->
+        []
+    end
+  end
+
   defp get_resource_route(opts, request) do
-    case Keyword.get(opts, :ot_resource_route, :ignore) do
-      route when is_binary(route) ->
-        route
-
-      infer_fn when is_function(infer_fn, 1) ->
-        infer_fn.(request)
-
-      :infer ->
-        Agent.get(
-          __MODULE__,
-          fn
-            {:ok, infer_fn} when is_function(infer_fn, 1) ->
+    resource_route =
+      case Keyword.get(opts, :ot_resource_route, :ignore) do
+        route when is_binary(route) ->
+          route
+
+        infer_fn when is_function(infer_fn, 1) ->
+          infer_fn.(request)
+
+        :infer ->
+          case get_defaults() do
+            {infer_fn, _} when is_function(infer_fn, 1) ->
               infer_fn.(request)
+
+            _ ->
+              raise ArgumentError,
+                    "The configured :infer_route keyword option value must be a function with an arity of 1"
           end
-        )
 
-      :ignore ->
-        nil
+        :ignore ->
+          nil
 
-      _ ->
-        raise ArgumentError,
-              "The :ot_resource_route keyword option value must either be a binary, a function with an arity of 1 or the :infer or :ignore atom"
-    end
+        _ ->
+          raise ArgumentError,
+                "The :ot_resource_route keyword option value must either be a binary, a function with an arity of 1 or the :infer or :ignore atom"
+      end
```

Comment:
I wonder if it would be better to try to extract this `case` into a multi-clause function since it's getting kinda big, wdyt?

Reply:
Certainly, done!